cmd/kubectl-gadget/deploy.go

```diff
@@ -473,6 +490,107 @@ func createAffinity(client *kubernetes.Clientset) (*v1.Affinity, error) {
 	return affinity, nil
 }
 
+// applyConfigToConfgMap overrides the ConfigMap using the following order of precedence (highest to lowest):
+// 1. Flags (--gadgets-public-keys, --disallow-gadgets-pulling, etc.) [Deprecated]
+// 2. --set-configmap flag
+// 3. ConfigFile
+// 4. ConfigMap default values
+// 5. Flags default values
+func applyConfigToConfgMap(cm *v1.ConfigMap, configPath string, flags *pflag.FlagSet) error {
+	cfgData, ok := cm.Data[configYamlKey]
+	if !ok {
+		return fmt.Errorf("%q not found in ConfigMap %q", configYamlKey, cm.Name)
+	}
+
+	v := viper.New()
+	v.SetConfigType("yaml")
+
+	// Set default value from flags
+	flags.VisitAll(func(flag *pflag.Flag) {
+		ck, ok := flagToConfigKey(flag.Name)
+		if !ok {
+			return
+		}
+		val := flagToConfigValue(ck, flag.Value)
+		v.SetDefault(ck, val)
+	})
+
+	// Set the default values from the ConfigMap
+	err := v.ReadConfig(strings.NewReader(cfgData))
+	if err != nil {
+		return fmt.Errorf("reading config file %q: %w", configPath, err)
+	}
+
+	// Set the values from a config file
+	if configPath != "" {
+		overrideCfg, err := os.ReadFile(configPath)
+		if err != nil {
+			return fmt.Errorf("reading override config file %q: %w", configPath, err)
+		}
+		err = v.MergeConfig(strings.NewReader(string(overrideCfg)))
+		if err != nil {
+			return fmt.Errorf("merging config file %q: %w", configPath, err)
+		}
+	}
+
+	// Set values from --set-configmap flag
+	for _, set := range setConfigmap {
+		parts := strings.SplitN(set, "=", 2)
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid format for --set-configmap: %q, expected key=value", set)
+		}
+		key := parts[0]
+		value := parts[1]
+		v.Set(key, value)
+	}
+
+	// Set values from flags if changed
+	flags.VisitAll(func(flag *pflag.Flag) {
+		fn, ok := flagToConfigKey(flag.Name)
+		if !ok || !flag.Changed {
+			return
+		}
+		val := flagToConfigValue(fn, flag.Value)
+		v.Set(fn, val)
+	})
+
+	var buf bytes.Buffer
+	err = v.WriteConfigTo(&buf)
+	if err != nil {
+		return fmt.Errorf("writing config to buffer: %w", err)
+	}
+	cm.Data[configYamlKey] = buf.String()
+
+	return nil
+}
+
+// flagToConfigKey converts a flag name to the corresponding config key
+// since some flags have different names in the config file.
+func flagToConfigKey(flagName string) (string, bool) {
+	key := flagName
+	switch flagName {
+	case "gadgets-public-keys":
+		key = gadgettracermanagerconfig.PublicKeys
+	case "disallow-gadgets-pulling":
+		key = gadgettracermanagerconfig.DisallowPulling
+	case "verify-gadgets":
+		key = gadgettracermanagerconfig.VerifyImage
+	}
+	if !gadgettracermanagerconfig.IsValidKey(key) {
+		return "", false
+	}
+
+	return gadgettracermanagerconfig.FullKeyPath(key), true
+}
+
+// flagToConfigValue converts a flag value to the corresponding config value.
+func flagToConfigValue(flagName string, value pflag.Value) any {
+	if flagName == gadgettracermanagerconfig.PublicKeys {
+		return strings.Split(value.String(), ",")
+	}
+	return value
+}
+
```

Comment:
Please comment that these are temporal helper functions to transform flags to configmap values and they have to be deleted once we remove flags.